src/Workspaces/Core/Portable/CodeGeneration/CodeGenerationOptions.cs

```diff
@@ -180,24 +178,24 @@ internal class CodeGenerationOptions
             this.ReuseSyntax = reuseSyntax;
 
             this.Options = options;
-            this.ParseOptions = parseOptions ?? this.BestLocation?.SourceTree.Options;
+            this.ParseOptions = parseOptions ?? this.BestLocation?.SourceTree?.Options;
         }
 
-        private static void CheckLocation(Location location, string name)
+        private static void CheckLocation(Location? location, string name)
         {
             if (location != null && !location.IsInSource)
             {
                 throw new ArgumentException(WorkspacesResources.Location_must_be_null_or_from_source, name);
             }
         }
 
-        internal Location BestLocation
+        internal Location? BestLocation
             => this.AfterThisLocation ?? this.BeforeThisLocation ?? this.ContextLocation;
 
         public CodeGenerationOptions With(
             Optional<Location> contextLocation = default,
-            Optional<Location> afterThisLocation = default,
-            Optional<Location> beforeThisLocation = default,
+            Optional<Location?> afterThisLocation = default,
```

Comment:
Man, optional of nullable? That seems weird, should it just be `Optional<Location>`?

Reply:
this is hte normal pattern for optional.  :)